File: python/012.py
def mergeTwoArr(arr1, arr2):
    mainArr = list(range(len(arr1)+len(arr2)))

    mainP, arr1P, arr2P = 0,0,0

    while arr1P < len(arr1) and arr2P < len(arr2):
        if arr1[arr1P] < arr2[arr2P]:
            mainArr[mainP] = arr1[arr1P]
            arr1P+=1
        else:
            mainArr[mainP] = arr2[arr2P]
            arr2P+=1
        mainP+=1

    while arr1P < len(arr1):
        mainArr[mainP] = arr1[arr1P]
        arr1P+=1
        mainP+=1

    while arr2P < len(arr2):
        mainArr[mainP] = arr2[arr2P]
        arr2P+=1
        mainP+=1

    return mainArr

print(mergeTwoArr([2,34,5,5,3],[3,4,42,2,3]))

File: golang/012.go
package main

import "fmt"

func mergeArr(arr1, arr2[]int) []int{
	mainArr := make([]int, len(arr1)+len(arr2))
	mainPointer := 0

	arr1Pointer := 0
	arr2Pointer := 0

	for arr1Pointer < len(arr1) && arr2Pointer < len(arr2){
		if arr1[arr1Pointer] < arr2[arr2Pointer]{
			mainArr[mainPointer] = arr1[arr1Pointer]
			arr1Pointer++
		} else {
			mainArr[mainPointer] = arr2[arr2Pointer]
			arr2Pointer++
		}
		mainPointer++
	}
	for arr1Pointer < len(arr1){

		mainArr[mainPointer] = arr1[arr1Pointer]
		arr1Pointer++
		mainPointer++
	}
	for arr2Pointer < len(arr2){
		mainArr[mainPointer] = arr2[arr2Pointer]
		arr2Pointer++
		mainPointer++
	}

	return mainArr
}

func main(){
	fmt.Println(mergeArr([]int{0, 7, 8},[]int{1,3,6}))
}